src/errno.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 Intel Corporation. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO: This module is not used now. Avoid the warning.
+#![allow(unused)]
+use std::fmt::{Display, Formatter};
+use std::io;
+use std::result;
+
+use libc::__errno_location;
+
+/// An error number, retrieved from [`errno`](http://man7.org/linux/man-pages/man3/errno.3.html),
+/// set by a libc function that returned an error.
+///
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Error(i32);
+pub type Result<T> = result::Result<T, Error>;
+
+impl Error {
+    /// Constructs a new error with the given `errno`.
+    ///
+    pub fn new(e: i32) -> Error {
+        Error(e)
+    }
+
+    /// Constructs an error from the current `errno`.
+    ///
+    /// The result of this only has any meaning just after a libc call that returned a value
+    /// indicating `errno` was set.
+    ///
+    pub fn last() -> Error {
+        Error(unsafe { *__errno_location() })
+    }
+
+    /// Gets the `errno` for this error.
+    ///
+    pub fn errno(&self) -> i32 {
+        self.0
+    }
+}
+
+impl Display for Error {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        write!(f, "Errno {}", self.errno())
+    }
+}
+
+impl From<io::Error> for Error {
+    fn from(e: io::Error) -> Self {
+        Error::new(e.raw_os_error().unwrap_or_default())
+    }
+}
+
+/// Returns the last `errno` as a [`Result`] that is always an error.
+///
+/// [`Result`]: type.Result.html
+///
```

Comment:
Should we remove the blank lines?

Reply:
I think blank line is no need here. The same as others.